src/config.ts

```diff
@@ -41,32 +41,58 @@ export class MirrordConfigManager {
     private constructor() {
         this.fileListeners = [];
 
-        this.fileListeners.push(vscode.workspace.onDidDeleteFiles(event => {
+        this.fileListeners.push(vscode.workspace.onDidDeleteFiles(async event => {
             const activeConfigDeleted = event.files.find(file => file.path === this.active?.path) !== undefined;
-            if (!activeConfigDeleted) {
+            if (activeConfigDeleted) {
+                new NotificationBuilder()
+                    .withMessage("removed active mirrord configuration")
+                    .withDisableAction("promptActiveConfigRemoved")
+                    .warning();
+
+                this.setActiveConfig(undefined);
+
                 return;
             }
 
-            new NotificationBuilder()
-                .withMessage("removed active mirrord configuration")
-                .withDisableAction("promptActiveConfigRemoved")
-                .warning();
+            if (this.active) {
+                try {
+                    await vscode.workspace.fs.stat(this.active);
+                } catch (e) {
+                    new NotificationBuilder()
+                        .withMessage("removed folder containing active mirrord configuration")
+                        .withDisableAction("promptActiveConfigRemoved")
+                        .warning();
 
-            this.setActiveConfig(undefined);
+                    this.setActiveConfig(undefined);
+                }
+            }
```

Comment:
Also here it might be nice to unify the cases, here it's more of a nitpick though, because it wouldn't significantly change the UX.